mayastor/src/grpc/v1/bdev.rs

```diff
@@ -0,0 +1,189 @@
+                }
+            } else {
+                Bdev::bdev_first()
+                    .into_iter()
+                    .for_each(|bdev| bdevs.push(bdev.into()));
+            }
+
+            Ok(ListBdevResponse {
+                bdevs,
+            })
+        })?;
+
+        rx.await
+            .map_err(|_| Status::cancelled("cancelled"))?
+            .map_err(Status::from)
+            .map(Response::new)
+    }
+
+    #[tracing::instrument(skip(self))]
+    async fn create(
+        &self,
+        request: Request<BdevRequest>,
+    ) -> Result<Response<BdevResponse>, Status> {
+        let uri = request.into_inner().uri;
+
+        let rx = rpc_submit(async move {
+            let name = bdev_create(&uri).await?;
+
+            if let Some(bdev) = Bdev::lookup_by_name(&name) {
+                Ok(Response::new(bdev.into()))
+            } else {
+                Err(NexusBdevError::BdevNotFound {
+                    name,
+                })
+            }
+        })?;
+
+        rx.await
+            .map_err(|_| Status::cancelled("cancelled"))?
+            .map_err(Status::from)
+    }
+
+    #[tracing::instrument(skip(self))]
+    async fn destroy(&self, request: Request<BdevRequest>) -> GrpcResult<()> {
+        let uri = request.into_inner().uri;
+
+        let rx = rpc_submit(async move { bdev_destroy(&uri).await })?;
+
+        rx.await
+            .map_err(|_| Status::cancelled("cancelled"))?
+            .map_err(Status::from)
+            .map(|_| Ok(Response::new(())))?
+    }
+
+    #[tracing::instrument(skip(self))]
+    async fn share(
+        &self,
+        request: Request<BdevShareRequest>,
+    ) -> Result<Response<BdevShareResponse>, Status> {
+        let r = request.into_inner();
+        let name = r.name;
+        let protocol = r.protocol;
+
+        if Bdev::lookup_by_name(&name).is_none() {
+            return Err(Status::not_found(name));
+        }
+
+        let bdev_name = name.clone();
+        let rx = match Protocol::try_from(protocol) {
+            Ok(Protocol::Nvmf) => {
+                rpc_submit::<_, String, CoreError>(async move {
+                    let bdev = Bdev::lookup_by_name(&bdev_name).unwrap();
+                    let share = bdev.share_nvmf(None).await?;
+                    let bdev = Bdev::lookup_by_name(&name).unwrap();
+                    Ok(bdev.share_uri().unwrap_or(share))
+                })
+            }
+
+            Ok(Protocol::Iscsi) => {
+                rpc_submit::<_, String, CoreError>(async move {
+                    let bdev = Bdev::lookup_by_name(&bdev_name).unwrap();
+                    let share = bdev.share_iscsi().await?;
+                    let bdev = Bdev::lookup_by_name(&name).unwrap();
+                    Ok(bdev.share_uri().unwrap_or(share))
+                })
+            }
+
+            Err(_) => {
+                return Err(Status::invalid_argument(protocol.to_string()))
+            }
+
+            _ => unreachable!(),
```

Comment:
Is this needed?

Reply:
For match to be exhaustive. Only iSCSI and NVMEoF are handled for bdev. If the Bdev share request comes with share protocol `Off`, we reach this state.